src/actions/user.js

```diff
@@ -40,12 +40,21 @@ export const updateUserSnippets = () => {
     };
     dispatch(updateUserSnippetsStarted());
     return axios.get(makeSaveEndpointUrl(username), { headers })
-      .then(
-        res => {
-          dispatch(setUserSnippets(res.data));
-          dispatch(updateUserSnippetsSucceeded());
-        },
-        err => dispatch(updateUserSnippetsFailed())
-      );
+      .then(res => {
+        // Jump to catch block if the user has no index.json file:
+        if (typeof(res.data) === 'string' &&
+            res.data.includes(
+          '<Error><Code>NoSuchKey</Code><Message>The specified key does not exist.</Message>'
+        )) {
```

Comment:
@Hopding I know I veered you towards this direction (you previously had those checks in place), so you can lambast me when I get in today